game/scripts/vscripts/heroes/antimage/antimage_spell_shield.lua

```diff
@@ -0,0 +1,179 @@
+end
+
+function lua_antimage_spell_shield:GetCooldown(level)
+    if self:GetCaster():HasScepter() then
+        return self:GetSpecialValueFor("scepter_cooldown")
+    end
+
+    return 0
+end
+
+
+-- Spell shield modifier
+modifier_lua_antimage_spell_shield = class({})
+
+function modifier_lua_antimage_spell_shield:IsHidden() return true end
+function modifier_lua_antimage_spell_shield:IsDebuff() return false end
+
+function modifier_lua_antimage_spell_shield:OnCreated()
+    -- Ability properties
+    self.caster = self:GetCaster()
+    self.ability = self:GetAbility()
+
+    -- Ability specials
+    self.spell_shield_resistance = self.ability:GetSpecialValueFor("spell_shield_resistance")    
+
+    -- Initialize table of old spells
+    self.caster.tOldSpells = {}
+    
+    if IsServer() then
+        -- Think, in order to check for scepter. Once a scepter is found, the modifier's interval starts
+        Timers:CreateTimer(function()            
+            if self.caster:HasScepter() then
+                self:StartIntervalThink(0.2)
+            else
+                return 20
+            end
+        end)
+    end
+end
+
+function modifier_lua_antimage_spell_shield:OnRefresh()
+    self:OnCreated()
+end
+
+-- Biggest thanks to Yunten and AtroCty
+-- Deleting old abilities
+function modifier_lua_antimage_spell_shield:OnIntervalThink()
+    if IsServer() then                
+
+        for i=#self.caster.tOldSpells,1,-1 do
+            local hSpell = self.caster.tOldSpells[i]
+            if hSpell:NumModifiersUsingAbility() == 0 and not hSpell:IsChanneling() then
+                hSpell:RemoveSelf()
+                table.remove(self.caster.tOldSpells,i)
+            end
+        end
+    end
+end
+
+function modifier_lua_antimage_spell_shield:DeclareFunctions() 
+    local decFuncs = {
+        MODIFIER_PROPERTY_MAGICAL_RESISTANCE_BONUS,
+        MODIFIER_PROPERTY_ABSORB_SPELL,
+        MODIFIER_PROPERTY_REFLECT_SPELL
+    }
+    return decFuncs 
+end
+
+function modifier_lua_antimage_spell_shield:GetModifierMagicalResistanceBonus()
+    if not self.caster:PassivesDisabled() then
+        return self.spell_shield_resistance
+    end
+end
+
+function modifier_lua_antimage_spell_shield:GetAbsorbSpell( params )
+    if IsServer() then    
+        if self.caster:HasScepter() and self.caster:IsRealHero() and self.ability:IsCooldownReady() then
+            if not self.caster:PassivesDisabled() then  
+                -- Apply Spell Absorption
+                local reflect_pfx = ParticleManager:CreateParticle("particles/units/heroes/hero_antimage/antimage_spellshield.vpcf", PATTACH_CUSTOMORIGIN_FOLLOW, self.caster)
+                ParticleManager:SetParticleControlEnt(reflect_pfx, 0, self.caster, PATTACH_POINT_FOLLOW, "attach_hitloc", self.caster:GetOrigin(), true)
+                ParticleManager:ReleaseParticleIndex(reflect_pfx)
+                return 1
+            end
+        end
+        return false  
```

Comment:
Either return true and false or 1 and 0, not 1 and false

Reply:
Agreed, copy pasted from imba's stuff.